src/python/WMComponent/TaskArchiver/CleanCouchPoller.py

```diff
@@ -380,6 +380,19 @@ def deleteWorkflowFromWMBSAndDisk(self):
         deletableWorkflowsDAO = self.daoFactory(classname="Workflow.GetDeletableWorkflows")
         deletablewfs = deletableWorkflowsDAO.execute()
 
+        # For T0 subtract the workflows which are not having all their blocks deleted yet:
+        if not self.useReqMgrForCompletionCheck:
+            deletedBlocksByWorkflowDAO = self.daoFactory(classname="Workflow.GetDeletedBlocksByWorkflow")
+            deletedBlocksByWorkflow = deletedBlocksByWorkflowDAO.execute()
+            for workflow in list(deletablewfs):
+                for ind in range(len(deletedBlocksByWorkflow)):
+                    if deletedBlocksByWorkflow[ind]['workflowName'] == workflow and \
+                       len(deletedBlocksByWorkflow[ind]['blocksNotDeleted']) != 0:
+                        msg = "Removing workflow: %s from the list of deletable workflows. It still has blocks NOT deleted."
+                        self.logger.debug(msg, workflow)
```

Comment:
If it doesn't pollute too much the T0 logs, it might be better to have an INFO log level here.

Reply:
done